content/source/docs/enterprise/system-overview/architecture.html.md

```diff
@@ -5,24 +5,202 @@ page_title: "Architecture Summary - System Overview - Terraform Enterprise"
+
+![tfe-int-webhook-receipt](assets/tfe-int-webhook-receipt.png)
+
+1. `ptfe_atlas` receives request from `ptfe_nginx`, pushes a job to Redis for `ptfe_sidekiq`, and then responds to the original request immediately.
+1. `ptfe_sidekiq` asynchronously pulls the job and passes the details to `ptfe_slug_ingress`.
+1. `ptfe_slug_ingress` spawns a separate `git clone` of your Git repository and `POST`s it to `ptfe_archivist`.
+1. `ptfe_archivist` persists the repository in blob storage and `POST`s the Object ID to `ptfe_atlas`.
+
+### Terraform Plan and Apply
+
+A state machine inside `ptfe_atlas` watches your workspaces at all times. The process is started when a plan or apply begins. The diagram below applies to both plan and apply stages.
+
+![tfe-int-tf-plan-apply](assets/tfe-int-tf-plan-apply.png)
+
+1. `ptfe_atlas` detects a state change and enqueues a background job for `ptfe_sidekiq` in Redis.
+2. `ptfe_sidekiq` asynchronously pulls the job and passes the details to `ptfe_build_manager`.
+3. `ptfe_build_manager` enqueues a `ptfe_build_worker` job in RabbitMQ.
+4. `ptfe_build_worker` spins up a temporary environment, runs a Terraform plan or apply, and passes the results to `ptfe_archivist`.
+5. `ptfe_archivist` persists the plan or apply results and returns a URL to `ptfe_build_worker`.
+6. `ptfe_build_worker` passes the plan or apply results back to `ptfe_build_manager`.
+7. `ptfe_build_manager` `POST`s the results to an internal webhook.
+8. `ptfe_nginx` receives the request and forwards the callback to `ptfe_atlas`.
+
+Finally, `ptfe_atlas` receives results of the plan or apply and begins next actions if any apply.
+
+### Sentinel Run
+
+A state machine inside `ptfe_atlas` is responsible for scheduling Sentinel runs when necessary.
+
+![tfe-int-sentinel-run](assets/tfe-int-sentinel-run.png)
+
+1. `ptfe_atlas` detects a state change and queues a background job for `ptfe_sidekiq` in Redis.
+1. `ptfe_sidekiq` asynchronously pulls the job and schedules a Sentinel run in `ptfe_nomad`.
+1. `ptfe_nomad` runs the Sentinel job and `POST`s the result back to `ptfe_atlas`.
+
+Although the result of the Sentinel run is known at this point, it is not persisted until the callback is processed:
+
+![tfe-int-sentinel-persistence](assets/tfe-int-sentinel-persistence.png)
+
+
+1. `ptfe_atlas` receives the callback request from `ptfe_nginx` and streams the response to `ptfe_archivist`.
+2. `ptfe_archivist` saves the Sentinel results to blob storage and returns an Object ID to `ptfe_atlas`.
+
+Finally, `ptfe_atlas` updates the metadata in PostgreSQL after it receives confirmation that the details of the run have been persisted.
+
+## Full Container List
+
+The following table lists every container Terraform Enterprise runs via the Replicated Native Scheduler.  The list is presented by layer as per the above document sections.
```

Comment:
so, this is all the long-running service containers, but not every container. there are a variety of ephemeral containers that run on startup and for batch jobs that will show up with random names.

Reply:
Is it the view of engineering that these non-randomly-named but ephemeral containers should be included in this internal doc?